### PR TITLE
Clean up `executable` package

### PR DESCRIPTION
**Pull request by basil**
Now that we're no longer supporting Java 8 for the `executable` package, some 50-odd lines of code can be deleted.

### Testing done

`mvn clean verify -DskipTests`

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: war/src/main/java/executable/Main.java [hunk 1]
```diff
@@ -109,16 +108,15 @@ public class Main {
...
-                                SUPPORTED_JAVA_VERSIONS));
+                System.err.printf(
+                        "Running with Java %d from %s, which is not fully supported. "
+                                + "Continuing because %s is set. "
+                                + "Supported Java versions are: %s. "
+                                + "See https://jenkins.io/redirect/java-support/ for more information.%n",
+                        releaseVersion,
+                        System.getProperty("java.home"),
+                        ENABLE_FUTURE_JAVA_CLI_SWITCH,
+                        SUPPORTED_JAVA_VERSIONS);
```
Comment (basil) [inline war/src/main/java/executable/Main.java:119]: Unnecessary use of `String.format`

File: war/src/main/java/executable/Main.java [hunk 2]
```diff
@@ -152,25 +150,6 @@ public class Main {
...
-        if (version.startsWith("1.")) {
-            String[] split = version.split("\\.");
-            if (split.length != 2) {
-                throw new NumberFormatException("Invalid Java specification version: " + version);
-            }
-            version = split[1];
-        }
-        return Integer.parseInt(version);
-    }
-
```
Comment (basil) [inline war/src/main/java/executable/Main.java:173]: Java 11 supports this natively.

File: war/src/main/java/executable/Main.java [hunk 3]
```diff
@@ -365,24 +349,6 @@ public static void main(String[] args) throws IllegalAccessException {
...
-        byte[] buffer;
-        try {
-            buffer = new byte[Math.min(in.available(), maxToRead)];
-            in.read(buffer);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        return new String(buffer);
-    }
-
```
Comment (basil) [inline war/src/main/java/executable/Main.java:385]: Java 11 supports this natively.

File: war/src/main/java/executable/Main.java [hunk 4]
```diff
@@ -443,21 +409,13 @@ public static File whoAmI(File directory) {
...
     }
 
-    private static void copyStream(InputStream in, OutputStream out) throws IOException {
-        byte[] buf = new byte[8192];
-        int len;
-        while ((len = in.read(buf)) > 0) {
-            out.write(buf, 0, len);
-        }
-    }
-
```
Comment (basil) [inline war/src/main/java/executable/Main.java:460]: Java 11 supports this natively.

File: war/src/main/java/executable/Main.java [hunk 5]
```diff
@@ -524,7 +482,7 @@ private static void deleteWinstoneTempContents(File file) {
 
     /**
      * Determines the home directory for Jenkins.
-     *
```
Comment (basil) [inline war/src/main/java/executable/Main.java:527]: Javadoc doesn't skip a line without `<p>`.

**Review by NotMyFault — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

**Pull request merged**